src/main/java/de/retest/recheck/report/TestReportFilter.java

```diff
@@ -78,20 +79,21 @@ static StateDifference filter( final StateDifference stateDifference, final Filt
 
 	static List<RootElementDifference> filter( final List<RootElementDifference> rootElementDifferences,
 			final Filter filter ) {
-		final List<RootElementDifference> newRootElementDifferences = new ArrayList<>();
-		for ( final RootElementDifference rootElementDifference : rootElementDifferences ) {
-			newRootElementDifferences.add( filter( rootElementDifference, filter ) );
-		}
-		return newRootElementDifferences;
+		return rootElementDifferences.stream() //
+				.map( rootElementDifference -> filter( rootElementDifference, filter ) ) //
+				.flatMap( newRootElementDifference -> newRootElementDifference.map( Stream::of ) //
+						.orElseGet( Stream::empty ) ) //
+				.collect( Collectors.toList() );
 	}
 
-	static RootElementDifference filter( final RootElementDifference rootElementDifference, final Filter filter ) {
-		final ElementDifference newElementDifference = filter( rootElementDifference.getElementDifference(), filter );
-		return new RootElementDifference( newElementDifference, rootElementDifference.getExpectedDescriptor(),
-				rootElementDifference.getActualDescriptor() );
+	static Optional<RootElementDifference> filter( final RootElementDifference rootElementDifference,
+			final Filter filter ) {
+		return filter( rootElementDifference.getElementDifference(), filter ) //
+				.map( newRootElementDifference -> new RootElementDifference( newRootElementDifference,
```

Comment:
Isn't `newRootElementDifference` a `newElementDifference`?
```suggestion
				.map( newElementDifference -> new RootElementDifference( newElementDifference,
```